broadcast/src/linked/signer/actor.rs

```diff
@@ -0,0 +1,939 @@
+        let result = Self {
+            runtime,
+            crypto: cfg.crypto,
+            hasher: cfg.hasher,
+            coordinator: cfg.coordinator,
+            application: cfg.application,
+            collector: cfg.collector,
+            chunk_namespace: encoder::chunk_namespace(&cfg.namespace),
+            ack_namespace: encoder::ack_namespace(&cfg.namespace),
+            refresh_epoch_timeout: cfg.refresh_epoch_timeout,
+            refresh_epoch_deadline: None,
+            rebroadcast_timeout: cfg.rebroadcast_timeout,
+            rebroadcast_deadline: None,
+            epoch_bounds: cfg.epoch_bounds,
+            mailbox_receiver,
+            journal_entries_per_section: cfg.journal_entries_per_section,
+            journal_replay_concurrency: cfg.journal_replay_concurrency,
+            journal_naming_fn: cfg.journal_naming_fn,
+            journals: HashMap::new(),
+            tip_man: TipManager::default(),
+            ack_man: AckManager::default(),
+            epoch: 0,
+        };
+
+        (result, mailbox)
+    }
+
+    /// Runs the actor until the runtime is stopped.
+    ///
+    /// The actor will handle:
+    /// - Timeouts
+    ///   - Rebroadcasting Links
+    ///   - Pruning old evidence
+    /// - Mailbox messages from the application:
+    ///   - Broadcast requests
+    ///   - Ack requests
+    /// - Messages from the network:
+    ///   - Links
+    ///   - Acks
+    pub async fn run(
+        mut self,
+        chunk_network: (impl Sender, impl Receiver),
+        ack_network: (impl Sender, impl Receiver),
+    ) {
+        let (mut link_sender, mut link_receiver) = chunk_network;
+        let (mut ack_sender, mut ack_receiver) = ack_network;
+        let mut shutdown = self.runtime.stopped();
+
+        loop {
+            // Enter the epoch
+            self.refresh_epoch();
+
+            // Create deadline futures.
+            // If the deadline is None, the future will never resolve.
+            let rebroadcast = match self.rebroadcast_deadline {
+                Some(deadline) => Either::Left(self.runtime.sleep_until(deadline)),
+                None => Either::Right(futures::future::pending()),
+            };
+            let refresh_epoch = match self.refresh_epoch_deadline {
+                Some(deadline) => Either::Left(self.runtime.sleep_until(deadline)),
+                None => Either::Right(futures::future::pending()),
+            };
+
+            select! {
+                // Handle shutdown signal
+                _ = &mut shutdown => {
+                    debug!("Shutdown");
+                    for (_, journal) in self.journals.drain() {
+                        journal.close().await.expect("unable to close journal");
+                    }
+                    return;
+                },
+
+                // Handle rebroadcast deadline
+                _ = rebroadcast => {
+                    debug!("Timeout: Rebroadcast");
```

Comment:
You can add vars to this to avoid `:`.

For example, you can do `debug!(reason="timeout", "rebroadcast")`